### PR TITLE
List View Expand row

### PR DESCRIPTION
**Pull request by jgiardino**
## Description

Compared updates I made with updates June made, and merged our conflicts.

Related PR: https://github.com/patternfly/patternfly/pull/487/files
Related story: https://patternfly.atlassian.net/browse/PTNFLY-452
## Changes
- For specific details about CSS changes, refer to the callouts in the images below.
- Made updates requested in https://github.com/patternfly/patternfly/pull/487/files
- I investigated including the blue border below the selected node in compound expansion, but this requires modifying how padding is applied to the elements. This also results in a layout that's too tall in the case where the nodes wrap due to padding being applied to the individual nodes. It might be possible to implement this a different way, but this would require further investigation.
- I also did not modify the colors applied to the compound nodes when active and on hover, since I'm not sure if we want to support the case where icons are not included (i.e. if only icons have the color change and they aren't included, then nothing in the node has a color change)
## Rawgit and Before/After Images

Rawgit links are:
https://rawgit.com/jgiardino/patternfly/expandRow4-dist/dist/tests/list-view-compound-expansion.html
and
https://rawgit.com/jgiardino/patternfly/expandRow4-dist/dist/tests/list-view-simple-expansion.html#

Examples Before and After Updates
1. I modified the border styles so that left and right blue borders are applied to the expanded state.
2. I also modified the border styles to avoid the double border that occurs when one list row is expanded, and the next list row is expanded or selected.
3. I replaced the <ul> implementation with <dl class="dl-horizontal">

listviewcssupdates-before-callouts

listviewcssupdates-after-callouts
## PR checklist (if relevant)
- [x] **Cross browser**: works in IE9
- [x] **Cross browser**: works in IE10
- [x] **Cross browser**: works in IE11
- [x] **Cross browser**: works in Edge
- [x] **Cross browser**: works in Chrome
- [x] **Cross browser**: works in Firefox
- [x] **Cross browser**: works in Safari
- [x] **Cross browser**: works in Opera
- [x] **Responsive**: works in extra small, small, medium and large view ports.
- [x] **Preview the PR**: An image or a URL for designer to preview this PR is provided.

### CODE REVIEW

**Comment by dlabrecq** [comment]
The bottom border is missing on larger screens. Although, perhaps this can be fixed later?

screen shot 2016-10-12 at 11 05 06 am

File: tests/pages/list-view-compound-expansion.html [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+categories: [Layouts]
+css-extra: false
+layout: layout
+resource: true
+full-page: true
+title: List View - Compound Expansion
+url-js-extra: ['//cdnjs.cloudflare.com/ajax/libs/c3/0.4.11/c3.min.js', '//cdnjs.cloudflare.com/ajax/libs/d3/3.5.17/d3.min.js']
```
Comment (jeff-phillips-18) [inline tests/pages/list-view-compound-expansion.html:8]: I don't believe the compound expansion example uses C3/D3 so these can be removed.

**Comment by dlabrecq** [comment]
Also wanted to note that clicking on the kebab (or almost anywhere in that row) will collapse/expand the pane. I had expected to only be able to click the arrow toggle.

**Review by jeff-phillips-18 — APPROVED** [review]
One minor nit but LGTM.

**Comment by jeff-phillips-18** [comment]
The issue is the kebab hit area is so small. Attempting to open the menu often misses and instead expands/collapses. Same is true for row selection. In general the kebabs should have more padding to increase the hit zone.

**Comment by dlabrecq** [comment]
Yes, I may have clicked slightly off the kebab. However, I can click anywhere and expand/collapse the row. It was just a little confusing when I tried clicking the kebab and the row collapsed.

**Comment by priley86** [comment]
@jgiardino nice work.

@dlabrecq @jeff-phillips-18 agreed on the kebabs hit zone increasing.

**Comment by jgiardino** [comment]
I also agree with @dlabrecq and @jeff-phillips-18 about the hit zone for the kebab. I also think the checkbox is a small target area too, and should be fixed along with that.

@LHinson What do you think? And is that something that should require a separate story, or is that considered part of this story?

**Comment by srambach** [comment]
Usability-wise, I find the borders confusing, and also with the hover color and selected color there's a lot going on when I move the mouse around. But the implementation LGTM with the comments above about clicking on the entire area/kebab.

**Comment by LHinson** [comment]
Yeah, we have a bug already filed for this. https://patternfly.atlassian.net/browse/PTNFLY-1344

Let's keep it as a separate issue @jgiardino @dlabrecq @jeff-phillips-18. I'll move it up in priority to address though given we are starting to use these frequently.

**Comment by andresgalante** [comment]
@dlabrecq what browser are you in to see the missing line?
also, +1 to kebab @LHinson lets fix it soon, it's a bug.

**Comment by andresgalante** [comment]
@jgiardino thanks a lot, it looks great and @junezhang Thanks for setting the base for this to happen.
I am ok to merge it but we should consider fixing the kebab soon.

**Comment by dlabrecq** [comment]
I found the border issue with Firefox.

Sorry if the kebab is causing confusion? I want to make it clear that clicking anywhere in the row will cause it to expand/collapse -- not just the toggle button.

**Pull request merged**